### PR TITLE
Openstack enhancements

### PR DESCRIPTION
**Pull request by mmartinv**

### CODE REVIEW

File: pkg/controller/map/storage/handler/openstack/handler.go [hunk 1]
```diff
@@ -92,9 +130,42 @@ func (r *Handler) changed(models ...*openstack.Storage) {
 		for _, pair := range mp.Spec.Map {
 			ref := pair.Source
 			for _, ds := range models {
-				if ref.ID == ds.ID || strings.HasSuffix(ds.Path, ref.Name) {
```
Comment (ahadas) [inline pkg/controller/map/storage/handler/openstack/handler.go:95]: it's unrelated to this particular PR but how about renaming `mp` and `ds` to something more meaningful?

**Review by ahadas — COMMENTED** [review]
looks good, minor comments

File: pkg/controller/provider/container/openstack/model.go [hunk 2]
```diff
@@ -164,17 +177,89 @@ func (r *ProjectAdapter) GetUpdates(ctx *Context, lastSync time.Time) (updates [
...
+			updates = append(updates, updater)
+
+		default:
+			updater := func(tx *libmodel.Tx) (err error) {
+				m := &model.Image{
+					Base: model.Base{ID: image.ID},
+				}
+				err = tx.Get(m)
+				if err != nil {
+					if errors.Is(err, &NotFound{}) {
```
Comment (bennyz) [inline pkg/controller/provider/container/openstack/model.go:231]: should be
```suggestion
					if errors.Is(err, libmodel.NotFound) {
```
no? (same for the other models)

Comment (mmartinv) [reply pkg/controller/provider/container/openstack/model.go:231]: You are right, I guess this is part was copied from another provider. I will change it in the next push

**Review by bennyz — APPROVED** [review]

File: pkg/controller/map/storage/handler/openstack/handler.go (hunk 1 above)
Comment (mmartinv) [reply pkg/controller/map/storage/handler/openstack/handler.go:95]: Done

File: pkg/controller/provider/container/openstack/client.go [hunk 3]
```diff
@@ -201,12 +187,13 @@ func (r *Client) insecure() bool {
 }
 
 // List Servers.
-func (r *Client) list(resourceType ResourceType, listopts interface{}) (obj interface{}, err error) {
```
[the remark this answers is not in the document]
Comment (ahadas) [reply pkg/controller/provider/container/openstack/client.go:204]: ack, I find the previous way clearer but that's subjective of course

**Review by ahadas — APPROVED** [review]

**Comment by ahadas** [comment]
@mmartinv there's an unsigned commit - please fix that and rebase

**Pull request merged**